packages/core/content-manager/admin/src/components/LeftMenu.tsx

```diff
@@ -120,18 +117,7 @@ const LeftMenu = () => {
                     key={link.uid}
                     to={{
                       pathname: link.to,
-                      /**
-                       * We re-add the plugins query to the params available in the menu,
-                       * this means once you've changed the locale in the app, you continue
-                       * to see the same locale when changing content-type.
-                       *
-                       * NOTE: if you go to a content-type that does not have i18n enabled,
-                       * we return the default documents anyway.
-                       */
-                      search: stringify({
-                        ...parse(link.search ?? ''),
-                        plugins: query.plugins,
-                      }),
+                      search: link.search ?? '',
```

Comment:
> I think the useful part of that code was to navigate while keeping the same locale selected when it's not the default one. For exemple, I have english and french, english is the default, after selecting french once I can cycle through french entries easily. Not groundbreaking but not completely useless either.

Thanks for clarifying I completely misunderstood the comment

> Also, we need to keep in mind that the plugins object could be used for things other than i18n.

Good catch I totally overlooked this!

Reply:
That's only me, but we might be mixing preferences and user flow/navigation here.

- If I prefer to see all my content in a different, non-default language, then it should be something I set in my preference and is saved in a persistent state (localStorage, DB, etc...)
- When navigating from content type to content type, it could make sense to try and persist the current locale, however, it might interfere with the preferred one.

I would suggest not trying to over-engineer a perfect solution and rather
1. If possible (from one localized CT to another one) pass the parameter, else lose it
2. Never transmit it + in the future use the user-preferred language instead

Reply:
Yes I thought the same last night that if we wanted to maintain a preferred language (besides the default) then it would need to be stored in some persistent state. That's more or less what I meant by "I think it will be difficult", in other words not something we should handle here.

I agree the question for now is do we carry it over when possible or always fallback to the default locale, and maybe that's a question for product.

Reply:
cc @yanniskadiri

Reply:
I agree we should ask product before trying to make it smarter. For now I think your solution is good though :)